internal/app/apid/main.go

```diff
@@ -215,3 +221,23 @@ func apidMain() error {
 
 	return errGroup.Wait()
 }
+
+func verifyExtKeyUsage(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+	if len(verifiedChains) == 0 {
+		return fmt.Errorf("no verified chains")
+	}
+
+	certs := verifiedChains[0]
```

Comment:
does this always have a slice of slice with exactly one cert?

Reply:
we check for chains length above, so at least one chain is there (it should always be there, as we require client cert)

so we walk through one chain and make sure there's a leaf cert which has proper key usage.

in practice there should be a single chain

Reply:
> in practice there should be a single chain

should we fail for some reason if there's more than one?

Reply:
I think if there's a single verified chain which has proper client cert, we should consider that to be positive check result - it means that client presented at least a single valid client certificate with proper key usage; if there's another chain, it doesn't matter if it has even wrong cert, as there's a valid one